Assets/Plugins/Editor/Uplift/DependencyResolution/Resolver.cs

```diff
@@ -0,0 +1,255 @@
+// --- BEGIN LICENSE BLOCK ---
+/*
+ * Copyright (c) 2019-present WeWantToKnow AS
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+// --- END LICENSE BLOCK ---
+
+
+using Uplift.Common;
+using Uplift.Packages;
+using Uplift.Schemas;
+using System.Collections.Generic;
+using UnityEngine;
+using System.Text;
+
+namespace Uplift.DependencyResolution
+{
+	class Resolver : IDependencySolver
+	{
+		public PackageList packageList;
+		DependencyGraph baseGraph = new DependencyGraph();
+		LinkedList<DependencyDefinition> originalDependencies;
+		Stack<State> stateStack = new Stack<State>();
+
+		public Resolver(DependencyGraph baseGraph, PackageList packageList)
+		{
+
+			this.baseGraph = baseGraph;
+			this.packageList = packageList;
+			if (packageList.GetType() == typeof(PackageListStub))
+			{
+				packageList.SetPackages(((PackageListStub)packageList).GetAllPackageRepo());
+			}
+		}
+
+		public Resolver(PackageList packageList)
+		{
+			this.packageList = packageList;
+		}
+
+		public void pushInitialState(DependencyDefinition[] dependencies, DependencyGraph dg)
+		{
+			Debug.Log("Pushing initial state");
+
+			this.originalDependencies = new LinkedList<DependencyDefinition>(dependencies);
+
+			//Create nodes for original dependencies
+			foreach (DependencyDefinition requested in originalDependencies)
+			{
+				DependencyNode node;
+				if (!dg.Contains(requested.Name))
+				{
+					node = new DependencyNode(requested);
+					dg.AddNode(node);
+				}
+				else
+				{
+					node = dg.FindByName(requested.Name);
+				}
+
+				if (node.restrictions.ContainsKey("initial"))
+				{
+					throw new IncompatibleRequirementException("initial requirements cannot have twice the same requirement");
+				}
+				else
```

Comment:
unecessary else indentation

Reply:
Not sure what you're referring to. C# conventions are not Java's 😄 

https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/if-else

Reply:
For @scassard I meant that you can write
```
 if (x)
 {
    throw ...
 }
 List<Con...
```
and avoid one indentation level. I tend to prefer this (for exceptions or `break/continue` statements) as it makes better use of the editor space.